src/mongocxx/lib/mongocxx/v_noabi/mongocxx/private/libbson.cpp

```diff
@@ -31,20 +31,41 @@ void doc_to_bson_t(const bsoncxx::v_noabi::document::view& doc, bson_t* bson) {
 
 }  // namespace
 
-scoped_bson_t::scoped_bson_t(bsoncxx::stdx::optional<bsoncxx::document::view_or_value> doc)
-    : _is_initialized{doc} {
+scoped_bson_t::scoped_bson_t(bsoncxx::document::view_or_value doc)
+    : _is_initialized{true}, _doc{std::move(doc)} {
+    doc_to_bson_t(*_doc, &_bson);
+}
+
+void scoped_bson_t::init_from_static(bsoncxx::document::view_or_value doc) {
+    _is_initialized = true;
+    _doc = std::move(doc);
+    doc_to_bson_t(*_doc, &_bson);
+}
+
+scoped_bson_t::scoped_bson_t(bsoncxx::document::view doc)
+    : scoped_bson_t(bsoncxx::document::view_or_value(doc)) {}
+
+void scoped_bson_t::init_from_static(bsoncxx::document::view doc) {
+    this->init_from_static(bsoncxx::document::view_or_value(doc));
+}
+
+scoped_bson_t::scoped_bson_t(bsoncxx::document::value doc)
+    : scoped_bson_t(bsoncxx::document::view_or_value(std::move(doc))) {}
+
+void scoped_bson_t::init_from_static(bsoncxx::document::value doc) {
+    this->init_from_static(bsoncxx::document::view_or_value(std::move(doc)));
+}
+
+scoped_bson_t::scoped_bson_t(bsoncxx::stdx::optional<bsoncxx::document::view_or_value> doc) {
     if (doc) {
-        _doc = std::move(doc);
-        doc_to_bson_t(*_doc, &_bson);
+        this->init_from_static(std::move(*doc));
     }
 }
```

Comment:
Is there an explanation on why these constructors needed to be revived? I'm guessing that one or more of the optional impls doesn't "do the right thing"?

Reply:
Correct. Rather than dealing with inconsistent `optional<T>` converting constructor behavior across configurations, settled on making it definitively unambiguous on all implementations for all expected view/value/view_or_value/optional<view_or_value> arguments expected to be given.